Lib/csharp/csharp.swg

```diff
@@ -82,8 +82,8 @@ SWIGINTERN const char * SWIG_UnpackData(const char *c, void *ptr, size_t sz) {
 %typemap(ctype) unsigned short,     const unsigned short &     "unsigned short"
 %typemap(ctype) int,                const int &                "int"
 %typemap(ctype) unsigned int,       const unsigned int &       "unsigned int"
-%typemap(ctype) long,               const long &               "int"
-%typemap(ctype) unsigned long,      const unsigned long &      "unsigned int"
+%typemap(ctype) long,               const long &               "long"
+%typemap(ctype) unsigned long,      const unsigned long &      "unsigned long"
```

Comment:
Why are the long types mapped to name "int" instead of "long"? Is that sensible?

Reply:
C int is 32-bits and so now it matches the C# 32-bit int. It is purposefully marshalled as 32-bits on both sides by default.

Reply:
Maybe I'm confused now, but the current swig master is the top version, which maps `long` to `int`. Is that correct?

Reply:
That's right. The default is to support C long as 32-bit. C long is sometimes 64-vit and sometimes 32-bit, whereas C int is always 32-bit. By using C int, it ensures the types used on both the unmanaged and managed sides are always 32-bit. SWIGWORDSIZE64 uses int instead of long long which is always 64-bit. Please see the updated CSharp.html docs in  fb91d1fa25ba00f93b9870d84ec3cbc0739650d7 and if not clear, improvements are welcome.

Reply:
I'm super sorry but I'm still confused what is the correct way to work with `size_t` and `ptrdiff_t` in current swig master. I have in my interface file:
```
#if defined(SWIGWORDSIZE64)
%apply unsigned long { size_t };
%apply const unsigned long& { const size_t& };
%apply long { ptrdiff_t };
%apply const long& { const ptrdiff_t& };
#else
%apply unsigned long long { size_t };
%apply const unsigned long long& { const size_t& };
%apply long long { ptrdiff_t };
%apply const long long& { const ptrdiff_t& };
#endif

// Include common stl types:
%include "stdint.i"
%include "std_string.i"
%include "std_shared_ptr.i"
%include "bda/StdContainers.i"

[...]
```

I'm under the impression that this generates an interface now where `size`t is mapped to `unsigned long long` even on platforms where `SWIGWORDSIZE64` is defined. I would expect it to use `unsigned long` for `size_t`, as the `%apply` directive says. I tested that `SWIGWORDSIZE64` is defined in the interface file, so this is not the cause of the problem. Why the `unsigned long long` instead of `unsigned long`?

The final code still works because on Linux, `unsigned long long` and `unsigned long` are both 64bit unsigned integral types. But the whole thing is confusing for me, sorry for that. Can you kindly help?